charts/prometheus-node-exporter/templates/service.yaml

```
@@ -19,6 +19,9 @@ spec:
   externalTrafficPolicy: {{ .Values.service.externalTrafficPolicy }}
 {{- end }}
   type: {{ .Values.service.type }}
+{{- if eq .Values.service.type "ClusterIP" }}
+  clusterIP: "{{ .Values.service.clusterIP }}"
+{{- end }}
```

I'd suggest to consider only user supplied (non-null) values. Otherwise, we should leave the field's value unspecified.

```suggestion
{{- if and (eq .Values.service.type "ClusterIP") .Values.service.clusterIP }}
  clusterIP: "{{ .Values.service.clusterIP }}"
{{- end }}
```